core/modules/wcontrol/Foreman.h

```diff
@@ -72,38 +74,17 @@ namespace lsst {
 namespace qserv {
 namespace wcontrol {
 
-class RunnerMgr;
-
-/// An abstract interface. Runners receive a reference to an
-/// object implementing this and make calls to report start and
-/// finish events for tasks they run. Schedulers must return
-/// TaskWatcher objects so that runners can make reports.
-class TaskWatcher {
-public:
-    virtual ~TaskWatcher() {}
-    virtual void markStarted(wbase::Task::Ptr t) {}
-    virtual void markFinished(wbase::Task::Ptr t) {}
-};
-
 /// An abstract scheduler interface. Foreman objects use Scheduler instances
 /// to determine what tasks to launch upon triggering events.
-class Scheduler : public TaskWatcher, public wbase::TaskScheduler {
+class Scheduler : public wbase::TaskScheduler, public util::CommandQueue {
 public:
     using Ptr = std::shared_ptr<Scheduler>;
     virtual ~Scheduler() {}
 
-    virtual bool removeByHash(std::string const& hash) { return false; }
-
     /// Take appropriate action when a task in the Schedule is cancelled. Doing
     /// nothing should be harmless, but some Schedulers may work better if cancelled
     /// tasks are removed.
     virtual void taskCancelled(wbase::Task *task) { return; }
-    virtual void queueTaskAct(wbase::Task::Ptr incoming) = 0;
-    virtual wbase::TaskQueuePtr nopAct(wbase::TaskQueuePtr running) = 0;
-    virtual wbase::TaskQueuePtr newTaskAct(wbase::Task::Ptr incoming,
-                                           wbase::TaskQueuePtr running) = 0;
-    virtual wbase::TaskQueuePtr taskFinishAct(wbase::Task::Ptr finished,
-                                              wbase::TaskQueuePtr running) = 0;
 };
 
 /// Foreman is a pooling thread manager that is pluggable with different scheduling objects.
```

Comment:
Oh, also, I think the big explanatory comment at the head of the file is also now stale -- would you please revise?